class-tgm-plugin-activation.php

```diff
@@ -758,17 +775,17 @@ public function notices() {
 					$install_link = true; // We need to display the 'install' action link.
 					$install_link_count++; // Increment the install link count.
 					if ( current_user_can( 'install_plugins' ) ) {
-						if ( isset( $plugin['required'] ) && $plugin['required'] ) {
-							$message['notice_can_install_required'][] = $plugin['name'];
+						if ( true === $plugin['required'] ) {
```

Comment:
I've added some type casting to the registration method now to account for this. Using some of the filter/filter emulate code I pointed to elsewhere from the WPSEO plugin. Not giving credit to them as I wrote it ;-)